### PR TITLE
backend/feat: #1361 on_status improving rebase

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Earlier `on_status` api used for sync only `booking` and ride `statuses`. After this PR we can do via `on_status`:

- update the same bap `ride` fields as we update via `on_update` (when bpp `ride.status` is not the same as bap `ride.status`)
- create bap `ride` if it is not there
- create `cancellation_reason` and `fare_breakup`s on bap side

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [x] This PR contains API breaking changes

## How did you test it?
Tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/dev/rest-client/dynamic-offer-driver-app/auto-flow.http [hunk 1]
```diff
@@ -1,6 +1,11 @@
 @token1 = {{driver-offer-bpp-auto-token1}}
```
Comment (khuzema786) [inline Backend/dev/rest-client/dynamic-offer-driver-app/auto-flow.http:1]: @roman-bodavskiy are this changes required to be pushed ?

File: Backend/dev/migrations/mock-registry/0000-init.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-- duplicated from seed, app is not running without migration
+ALTER TABLE atlas_registry.subscriber OWNER TO atlas_registry_user;
```
Comment (khuzema786) [inline Backend/dev/migrations/mock-registry/0000-init.sql:2]: This also is it required to be pushed ?

File: Backend/dev/rest-client/dynamic-offer-driver-app/auto-flow.http (hunk 1 above)
Comment (roman-bodavskiy) [reply Backend/dev/rest-client/dynamic-offer-driver-app/auto-flow.http:1]: this changes affect only local testing with rest client vs code extension, I was not able to test without them, so can be pushed I think

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs [hunk 3]
```diff
@@ -11,42 +11,256 @@
...
+                  rideStartTime = Just rideStartedInfo.rideStartTime,
+                  driverArrivalTime = rideStartedInfo.driverArrivalTime
+                 }
+    RideCompletedDetails {newRideInfo, rideStartedInfo, rideCompletedInfo} -> do
+      rideEntity <- buildRideEntity booking updateRideCompleted newRideInfo
+      rideBookingTransaction bookingNewStatus rideNewStatus booking rideEntity
+      when (isStatusChanged booking.status bookingNewStatus rideEntity) $ do
+        breakups <- traverse (buildFareBreakup booking.id) rideCompletedInfo.fareBreakups
+        QFareBreakup.deleteAllByBookingId booking.id
+        QFareBreakup.createMany breakups
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs:179]: Also here we are deleting FareBreakups by booking.id and then creating new ones. Can we just update the existing fare breakups in this case instead of deleting and creating new ones.

File: Backend/dev/migrations/mock-registry/0000-init.sql (hunk 2 above)
Comment (roman-bodavskiy) [reply Backend/dev/migrations/mock-registry/0000-init.sql:2]: This is also required for local testing. Mock-registry failed to run locally because of empty directory `dev/migrations/mock-registry.` 
If it bothers us, I can remove

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs (hunk 3 above)
Comment (roman-bodavskiy) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs:179]: I was thinking about that. then we need to update each breakup by description (id generated on bap side, did not come from bpp). What if changed number of breakups, or descriptions? logic will be too complicated

Anyway this logic works when status changed, so most likely we will not delete anything, only create

Comment (khuzema786) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs:179]: Sure @roman-bodavskiy Works

File: Backend/dev/migrations/mock-registry/0000-init.sql (hunk 2 above)
Comment (khuzema786) [reply Backend/dev/migrations/mock-registry/0000-init.sql:2]: Ohh yes, someone faced this issue too, we just created a directory to fix it. Sure we can have this local fixes to solve such issues

**Review by khuzema786 — APPROVED** [review]
